crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -224,31 +226,79 @@ where
         &mut self,
         input: &Message,
     ) -> Result<Vec<Message>, ConversionError> {
-        let tedge_event = ThinEdgeEvent::try_from(input.topic.name.as_str(), input.payload_str()?)?;
-        let c8y_event = C8yCreateEvent::try_from(tedge_event)?;
+        let mut vec: Vec<Message> = Vec::new();
+        let c8y_event;
+        // check if there is a childid in the topic, if not create the child before forwarding the event message
+        let child_id: Option<String> = get_child_id_from_event_topic(&input.topic.name);
+
+        let message = match child_id {
+            Some(ref c_id) => {
+                if !self.children.contains(c_id) {
+                    self.children.insert(c_id.clone());
+                    vec.push(Message::new(
+                        &Topic::new_unchecked(SMARTREST_PUBLISH_TOPIC),
+                        format!("101,{c_id},{c_id},thin-edge.io-child"),
+                    ));
+                }
 
-        // If the message doesn't contain any fields other than `text` and `time`, convert to SmartREST
-        let message = if c8y_event.extras.is_empty() {
-            let smartrest_event = Self::serialize_to_smartrest(&c8y_event)?;
-            let smartrest_topic = Topic::new_unchecked(SMARTREST_PUBLISH_TOPIC);
+                let tedge_event =
+                    ThinEdgeEvent::try_from(&input.topic.name, input.payload_str()?, Some(c_id))?;
 
-            Message::new(&smartrest_topic, smartrest_event)
-        } else {
-            // If the message contains extra fields other than `text` and `time`, convert to Cumulocity JSON
-            let cumulocity_event_json = serde_json::to_string(&c8y_event)?;
-            let json_mqtt_topic = Topic::new_unchecked(C8Y_JSON_MQTT_EVENTS_TOPIC);
+                c8y_event = C8yCreateEvent::try_from(tedge_event)?;
+                // If the message contains extra fields other than `text` and `time`, convert to Cumulocity JSON
```

Comment:
Is this comment applicable for child device case? From your code, it doesn't look like that.

Reply:
Yes, that's right. It's not applicable here.